### PR TITLE
planner: fix same index read plan but has different plan digest

### LINKED ISSUE #31438: Normalized plan is not following the same format
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Execute any workload with a coprocessor push down.

### 2. What did you expect to see? (Required)

For the normalized plan, we should know that the push down happens for TiKV or TiFlash, like:

```
	StreamAgg_17           	root     	...
	└─IndexReader_18       	root     	...
	  └─StreamAgg_9        	cop[tikv]	...
	    └─IndexRangeScan_16	cop[tikv]	...
```

### 3. What did you see instead (Required)

No TiKV or TiFlash push down information is included in the normalized plan:

```
	StreamAgg           	root	funcs:count(?)->?
	└─IndexReader       	root	index:StreamAgg_9
	  └─StreamAgg       	cop 	funcs:count(test.sbtest1.k)->?
	    └─IndexRangeScan	cop 	table:sbtest1, index:k_1(k), range:[?,?], keep order:false
```

Note that this is not the first time that a plan output is not displayed in the same way:

https://github.com/pingcap/tidb/issues/28731

### 4. What is your TiDB version? (Required)

v5.4.0

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Issue Number: close #31438

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

N/A

### Release note

```release-note
None
```

### CODE REVIEW

File: planner/core/explain.go [hunk 1]
```diff
@@ -431,7 +431,7 @@ func (p *PhysicalIndexReader) ExplainInfo() string {
 
 // ExplainNormalizedInfo implements Plan interface.
 func (p *PhysicalIndexReader) ExplainNormalizedInfo() string {
```
Comment (breeswish) [inline planner/core/explain.go:433]: Can you provide tests to verify and ensure that people in future won't change it by mistake?

Comment (crazycs520) [reply planner/core/explain.go:433]: I have already add a failpoint-test here, I already tested without this change, some plan normalized test will be failed.

**Review by breeswish — APPROVED** [review]

**Pull request merged**